File: wc.py
#!/usr/bin/env python3

import time


def is_white_space(c):
    return c == '\r' or c == '\n' or c == ' ' or c == '\t'


def word_count(file_path: str):
    with open(file_path, "rt") as file:
        str = file.read()

        wc = 0
        is_prev_white_space = True
        for i in range(1, len(str)):
            is_cur_white_space = is_white_space(str[i])

            if is_cur_white_space and not is_prev_white_space:
                wc += 1
            is_prev_white_space = is_cur_white_space

        if len(str) != 0 and not is_prev_white_space:
            wc += 1

        print(f'Num Words = {wc}\n')


start = time.time()
word_count("file.txt")
end = time.time()
elapsed = (end - start) * 1000
print(f"Elapsed = {elapsed} ms\n")

File: wc.go
package main

import (
	"fmt"
	"os"
	"time"
)

func isWhiteSpace(c rune) bool {
	return c == '\r' || c == '\n' || c == ' ' || c == '\t'
}

func WordCount(str string) {
	wc := 0
	is_prev_white_space := false

	for i := 0; i < len(str); i++ {
		is_cur_white_space := isWhiteSpace(rune(str[i]))

		if is_cur_white_space && !is_prev_white_space {
			wc += 1
		}

		is_prev_white_space = is_cur_white_space
	}

	if len(str) != 0 && !is_prev_white_space {
		wc += 1
	}

	fmt.Printf("Num Words = %d\n", wc)
}

func main() {
	start := time.Now()

	b, err := os.ReadFile("file.txt")
	if err != nil {
		fmt.Print(err)
	}

	str := string(b) // convert content to a 'string'
	WordCount(str)

	elapsed := time.Since(start)
	fmt.Printf("Elapsed = %d ms\n", elapsed.Milliseconds())
}
